### PR TITLE
Update PyPI Integrations for Release Upload

### LINKED ISSUE #71: Update PyPI Integration
With recent changes to PyPI trusted publishers, we need to update the CI to use that instead of the old non-functional username / password setup.

Update the integration so that releases may publish to PyPI again.

### PR DESCRIPTION
**Pull request by JN-Hernandez**
## Overview

The `python-omgeo` Python packages is published via the Azavea PyPI account, however recent builds of these projects are failing because PyPI has changed from username/password authentication to token based and trusted-publisher based authentication.  This PR will address updating the integration.

Closes #71 

### Demo

The Problem:
- [GitHub Action Release #5](https://github.com/azavea/python-omgeo/actions/runs/7631232087)
```
Uploading python_omgeo-6.2.1-py3-none-any.whl
25l
  0% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/47.7 kB • --:-- • ?
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 47.7/47.7 kB • 00:00 • 145.3 MB/s
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 47.7/47.7 kB • 00:00 • 145.3 MB/s
25hWARNING  Error during upload. Retry with the --verbose option for more details. 
ERROR    HTTPError: 403 Forbidden from https://upload.pypi.org/legacy/          
         Username/Password authentication is no longer supported. Migrate to API
         Tokens or Trusted Publishers instead. See                              
         https://pypi.org/help/#apitoken and                                    
         https://pypi.org/help/#trusted-publishers                              
Error: Process completed with exit code 1.
```

Proposed Solution:
- [django-ecsmanage PyPI Integration](https://github.com/azavea/django-ecsmanage/blob/cee8ec1b361fae37f941381118d900010cf3478d/.github/workflows/release.yml#L33)
```
      - name: Upload release to TestPyPI
        uses: pypa/gh-action-pypi-publish@release/v1
```

### Notes

- Prior to merging this PR, TestPyPI and PyPI were updated to add a new publisher, as outlined in the workplan on Issue #71 
- After conducting a test upload to TestPyPI and confirming success, update the `workflow.yml` file by removing the `with:` clause from the **Upload release to TestPyPI** step and update the step name accordingly.  This will switch the upload from TestPyPI to PyPI.

## Testing Instructions

TestPyPI Testing - Manual Release:
> [!NOTE]
> Example of successful manual test: [GitHub Action Release #6](https://github.com/azavea/python-omgeo/actions/runs/8693618051/job/23840762347)
> Note that `upload` was misspelled in this test run, it has since been corrected.
- [x] 1. Create a manual test release branch off this one titled `release/test` with the tags `test`, then push up
```
git checkout -b release/test
git tag test
git push --set-upstream origin release/test
git push --tags
```
- [x] 2. Review the GitHub Action generated to see if it completes successfully
- [x] 3. Remove the tags from origin and the local branch
```
git tag -d test
git push --delete origin test
```

## Checklist

- [x] 1. `CHANGES.txt` has been updated
- [ ] 2. After merging, ensure a release workflow has been conducted

### CODE REVIEW

**Comment by JN-Hernandez** [comment]
CI fails a couple of tests:
```
test_impatiently_geocode_azavea (omgeo.tests.tests.GeocoderTest) ... ok

======================================================================
FAIL: test_geocode_esri_wgs_senado_mx (omgeo.tests.tests.GeocoderTest)
Attempt to geocode ``Paseo de la Reforma 135, Tabacalera,
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/python-omgeo/python-omgeo/omgeo/tests/tests.py", line 183, in test_geocode_esri_wgs_senado_mx
    self.assertEqual(search_text in candidates[0].match_addr, True,
AssertionError: False != True : "Paseo de la Reforma 135" not found in match_addr. Got "Paseo de La Reforma 135, Tabacalera, Ciudad de México, Cuauhtémoc, Ciudad de México, 06030".

======================================================================
FAIL: test_geocode_structured_esri_wgs_senado_mx (omgeo.tests.tests.GeocoderTest)
Attempt to geocode ``Paseo de la Reforma 135, Tabacalera,
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/python-omgeo/python-omgeo/omgeo/tests/tests.py", line 196, in test_geocode_structured_esri_wgs_senado_mx
    self.assertEqual(search_text in candidates[0].match_addr, True,
AssertionError: False != True : "Paseo de la Reforma 135" not found in match_addr. Got "Paseo de La Reforma 135, Tabacalera, Ciudad de México, Cuauhtémoc, Ciudad de México, 06030".

----------------------------------------------------------------------
Ran 53 tests in 5.887s

FAILED (failures=2, skipped=14)
Test failed: <unittest.runner.TextTestResult run=53 errors=0 failures=2>
error: Test failed: <unittest.runner.TextTestResult run=53 errors=0 failures=2>
Error: Process completed with exit code 1.
```

Uncertain if this is because the data being returned has changed or if a function isn't parsing out the data returned correctly.  Additional investigation needed.

**Comment by JN-Hernandez** [comment]
Confirmed with @rajadain that the data being returned changed for the affected tests and that we should simply update the tests to match expected return.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -25,10 +27,10 @@ jobs:
...
-        env:
-          TWINE_USERNAME: ${{ secrets.PYPI_AZAVEA_USERNAME }}
-          TWINE_PASSWORD: ${{ secrets.PYPI_AZAVEA_PASSWORD }}
-        run: |
-          python setup.py sdist bdist_wheel
-          twine upload dist/*
+      - name: Build release package
+        run: python setup.py sdist bdist_wheel
+
+      - name: Upload release to TestPyPI
```
Comment (rajadain) [inline .github/workflows/release.yml]: Please update this to point to production PyPI, not test

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -25,10 +27,10 @@ jobs:
...
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
```
Comment (rajadain) [inline .github/workflows/release.yml]: Same here

**Review by rajadain — CHANGES_REQUESTED** [review]
Nice work! But we need to update the GitHub Actions to point to  production PyPI before this can be merged.

**Review by rajadain — APPROVED** [review]
Looks good, nice work!

**Pull request merged**